systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java

```diff
@@ -0,0 +1,353 @@
+                        ResourceManager.kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME + "-tls").get(0).getMetadata().getName();
+
+                    InternalKafkaClient internalTlsKafkaClient = new InternalKafkaClient.Builder()
+                        .withUsingPodName(kafkaClientsTlsPodName)
+                        .withListenerName(listener.getName())
+                        .withTopicName(topicName)
+                        .withNamespaceName(NAMESPACE)
+                        .withClusterName(CLUSTER_NAME)
+                        .withKafkaUsername(kafkaUsername)
+                        .withMessageCount(MESSAGE_COUNT)
+                        .build();
+
+                    LOGGER.info("Checking produced and consumed messages to pod:{}", kafkaClientsTlsPodName);
+
+                    // verify phase
+                    internalTlsKafkaClient.checkProducedAndConsumedMessages(
+                        internalTlsKafkaClient.sendMessagesTls(),
+                        internalTlsKafkaClient.receiveMessagesTls()
+                    );
+                } else {
+                    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME + "-plain").done();
+
+                    final String kafkaClientsPlainPodName =
+                        ResourceManager.kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME + "-plain").get(0).getMetadata().getName();
+
+                    InternalKafkaClient internalPlainKafkaClient = new InternalKafkaClient.Builder()
+                        .withUsingPodName(kafkaClientsPlainPodName)
+                        .withListenerName(listener.getName())
+                        .withTopicName(topicName)
+                        .withNamespaceName(NAMESPACE)
+                        .withClusterName(CLUSTER_NAME)
+                        .withMessageCount(MESSAGE_COUNT)
+                        .build();
+
+                    LOGGER.info("Checking produced and consumed messages to pod:{}", kafkaClientsPlainPodName);
+
+                    // verify phase
+                    internalPlainKafkaClient.checkProducedAndConsumedMessages(
+                        internalPlainKafkaClient.sendMessagesPlain(),
+                        internalPlainKafkaClient.receiveMessagesPlain()
+                    );
+                }
+            }
+        }
+    }
+
+    private Map<KafkaListenerType, List<GenericKafkaListener>> generateTestCases() {
+
+        LOGGER.info("Starting to generate test cases for multiple listeners");
+
+        int stochasticCount;
+
+        for (KafkaListenerType kafkaListenerType : KafkaListenerType.values()) {
+
+            LOGGER.info("Generating {} listener", kafkaListenerType.name());
+
+            List<GenericKafkaListener> testCaseListeners = new ArrayList<>(5);
+
+            switch (kafkaListenerType) {
+                case NODEPORT:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 5);
```

Comment:
The problem with random tests is that they're not reproducible, so if you get a failure and think you've fixed it how do you re-run the same test again? The solution is in the fact that `java.util.Random` is a pseudo-random number generator and the contract says if you give it the same seed it will generate the same sequence. But got can't get the seed. So in normal running you can generate a seed using one `Random` (or use `System.nanoTime()` or whatever) and use it as the seed for another Random instance and _log the seed_. Then you can reproduce any failed test for which you have the logs by having some System property or env var which allows you to set the seed, rather than using a `Random`/`System.nanoTime()` whatever.

If we're going to use more random testing in the future this is a pattern we should encapsulate, but in another PR.